nbextensions/appCell2/widgets/appCellWidget.js

```diff
@@ -157,237 +157,235 @@ define([
 
         // TABS
 
-        function pRequire(module) {
-            return new Promise((resolve, reject) => {
-                require(module, (...args) => {
-                    resolve(args);
-                }, (err) => {
-                    reject(err);
-                });
-            });
-        }
-
         function loadParamsWidget(arg) {
-            return pRequire(['./appParamsWidget']).spread((Widget) => {
-                // TODO: widget should make own bus.
-                const widgetBus = runtime
-                        .bus()
-                        .makeChannelBus({ description: 'Parent comm bus for input widget' }),
-                    widget = Widget.make({
-                        bus: widgetBus,
-                        initialParams: model.getItem('params'),
+            return lang
+                .pRequire(['nbextensions/appCell2/widgets/appParamsWidget'])
```

Comment:
This is the main change that was made - use `lang.pRequire` instead of its own local copy. This is one of the spots that makes me scratch my head - if we already know the name and path to the module to load, why does it need to be loaded later rather than as a top level dependency? It doesn't really execute anything, and it's not like it's multiple MB of code. 🤷

Reply:
I've also scratched my head around this area, as well as doing a bit of

![head-desk-interaction](https://user-images.githubusercontent.com/556057/134017771-a1d0a8d3-ab63-44de-a867-352fef9f1d76.gif)

If we could shift over to using a similar pattern for the params edit/view as in the bulk import cell, some of this could just disappear.